### PR TITLE
Add API for accessing node logs in test clusters framework

### PR DESCRIPTION
**Pull request by mark-vieira**
Some tests need to assert on log messages in node logs. This adds an API for accessing test cluster node logs for this use case.

### CODE REVIEW

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/ClusterHandle.java [hunk 1]
```diff
@@ -127,4 +128,9 @@ public interface ClusterHandle extends Closeable {
      * Cleans up any resources created by this cluster.
      */
     void close();
+
+    /**
+     * Returns an {@link InputStream} for the given node log.
+     */
+    InputStream streamNodeLog(int index, LogType logType);
```
Comment (rjernst) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/ClusterHandle.java:135]: maybe just "getNodeLog"? It's not really "streaming" exactly

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.cluster;
+
+public enum LogType {
+    SERVER("elasticsearch.log"),
+    SERVER_JSON("elasticsearch_server.json");
```
Comment (rjernst) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java:13]: Don't these all assume a specific cluster name?

Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java:13]: I was wondering the same thing. Not when I tested it locally. Even with a cluster name of `test-cluster` the log file was just `elasticsearch.log`. What affects this behavior?

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/ClusterHandle.java (hunk 1 above)
Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/ClusterHandle.java]: Done.

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java (hunk 2 above)
Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java:13]: 🤦 I just now noticed we never actually set `cluster.name` in the new framework. Perhaps we should?

Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/LogType.java]: Ok, I've fixed this. We now set `cluster.name` and resolve the log filename appropriately.

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**